src/freenas/usr/local/sbin/migrate_warden.py

```diff
@@ -0,0 +1,400 @@
+
+            if prop == "DHCP":
+                print("  DHCP isn't supported by iocage currently, not"
+                      " migrating property")
+                continue
+
+            # If prop is empty, the 'prop' didn't exist in Warden
+            if prop:
+                props[ioc_prop] = prop
+
+        self.activate_pool()
+        running = self.is_jail_running()
+
+        if running:
+            print(f"  {self.jail} is running, please stop it first.")
+            return
+
+        self.create_jail(props)
+        # async_tasks = [
+        #     [self.ZFS.send_dataset, self.s_pipe, self.warden_dataset,
+        #      self.date],
+        #     [self.ZFS.recv_dataset, self.r_pipe, self.iocage_root]
+        # ]
+
+        await asyncio.gather(
+            asyncio.ensure_future(self.loop.run_in_executor(
+                self.thread_pool_executor,
+                functools.partial(
+                    self.ZFS.send_dataset,
+                    self.s_pipe,
+                    self.warden_dataset,
+                    self.date
+                )
+            )),
+            asyncio.ensure_future(self.loop.run_in_executor(
+                self.thread_pool_executor,
+                functools.partial(
+                    self.ZFS.recv_dataset,
+                    self.r_pipe,
+                    self.iocage_root
+                )
+            ))
+        )
+
+        # await asyncio.gather(
+        #     asyncio.ensure_future(self.loop.run_in_executor(
+        #         self.thread_pool_executor, functools.partial(*task)
+        #     ))
+        #     for task in async_tasks
+        # )
+
+        self.warden_dataset.destroy_snapshot(f"WardenMigration_{self.date}")
+        self.copy_fstab()
+
+    def is_jail_running(self):
+        """
+        :return: boolean if jail is running or not
+        """
+        cmd = ["jls", "--libxo", "json"]
+        jls_json = json.loads(su.check_output(cmd))["jail-information"]["jail"]
+
+        for jail in jls_json:
+            p = f"{self.dir}/{self.jail}"
+            _p = jail["path"]
+
+            if p == _p:
+                return True
+
+        return False
+
+    def jail_props(self, prop):
+        """
+        :return: the specified files contents
+        """
+        try:
+            with open(f"{self.meta}/{prop}", "r") as p:
+                _p = p.read().rstrip()
+
+                if "allow." in _p:
+                    single_period = ["allow_raw_sockets", "allow_socket_af",
+                                     "allow_set_hostname"]
+
+                    if _p in single_period:
+                        _p = _p.replace(".", "_", 1).replace("true", "1")
+                    else:
+                        _p = _p.replace(".", "_").replace("true", "1")
+                elif prop == "vnet":
+                    _p = "on"
+                elif prop == "autostart":
+                    _p = "on"
+                elif prop == "nat":
```

Comment:
This is used more frequently than you might think. If you have a primary interface on network 192.168.0.x/24, and you want your jails (or jails) to have 10.0.0.x/24 addresses, you will require NAT. It's nowhere near as common as the common case (DHCP), but it definitely exists and gets use.